### PR TITLE
[TST] : Adding test cases to confirm that .apply() works fine

### LINKED ISSUE #21535: Unexpected behaviour of apply function
#### Code Sample, a copy-pastable example if possible

```python
import numpy as np
import pandas as pd

tmp = np.array([[1,1,1], [1,0,0], [1,1,0], [1,1,0]])
tmp_df = pd.DataFrame(tmp)

tmp_df.apply(lambda x: x[x==1].index.tolist() , axis=1) # throws ValueError [I]
Out[0]:
ValueError: could not broadcast input array from shape (2) into shape (3)

tmp_df.apply(lambda x: [x[x==1].index.tolist()] , axis=1) # This doesn't throw Error [II]
Out[1]: 
0    [[0, 1, 2]]
1          [[0]]
2       [[0, 1]]
3       [[0, 1]]
dtype: object

tmp2 = np.array([[1,0,0], [1,0,0], [0,1,1]])
tmp2_df.apply(lambda x: x[x==1].index.tolist(), axis=1)  # This works
Out[2]: 
0       [0]
1       [0]
2    [1, 2]
dtype: object

```
#### Problem description
my pandas version is v0.22.0 and numpy version is 0.14.0
The expected output of [I] is a Series containing a list in each row but it's throwing an ValueError. 

#### Expected Output
[I] should return,
0    [0, 1, 2]
1          [0]
2       [0, 1]
3       [0, 1]

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.4.final.0
python-bits: 64
OS: Darwin
OS-release: 17.6.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8
pandas: 0.22.0
pytest: 3.4.0
pip: 9.0.1
setuptools: 38.5.1
Cython: 0.27.3
numpy: 1.14.0
scipy: 1.0.0
pyarrow: None
xarray: None
IPython: 6.2.1
sphinx: 1.7.0
patsy: 0.5.0
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: 1.2.1
tables: None
numexpr: 2.6.4
feather: None
matplotlib: 2.1.2
openpyxl: 2.5.0
xlrd: 1.1.0
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by uds5501**
Adding test cases for closure of issue  :  Unexpected behaviour of apply function 

- [x] closes #21535
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by uds5501** [comment]
@gfyoung  Please see if this is suffice for test case

**Comment by gfyoung** [comment]
@uds5501 : Good start!  Can you include all of the examples from that issue under your test?

File: pandas/tests/sparse/frame/test_apply.py [hunk 1]
```diff
@@ -90,3 +90,22 @@ def test_applymap(frame):
     # just test that it works
     result = frame.applymap(lambda x: x * 2)
     assert isinstance(result, SparseDataFrame)
+
```
Comment (jreback) [inline pandas/tests/sparse/frame/test_apply.py]: is there a reason you are putting them in the sparse tests?

these look quite duplicative of the current apply tests, which there are many.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/sparse/frame/test_apply.py (hunk 1 above)
Comment (uds5501) [reply pandas/tests/sparse/frame/test_apply.py]: @jreback well, there is no specific reason for putting them in sparse tests, as for duplicative tests, I initially committed only one test which threw error until it was recommended to add more

Comment (gfyoung) [reply pandas/tests/sparse/frame/test_apply.py]: @uds5501 : I was saying that you should add each example, but you didn't need to create separate tests with duplicate logic to do so.

Comment (uds5501) [reply pandas/tests/sparse/frame/test_apply.py]: @gfyoung  Kindly suggest a method, I mean I am little confused here by what adding each example exactly means

Comment (gfyoung) [reply pandas/tests/sparse/frame/test_apply.py]: @uds5501 : I think you actually figured out 😄

File: pandas/tests/sparse/frame/test_apply.py [hunk 2]
```diff
@@ -90,3 +91,18 @@ def test_applymap(frame):
...
+def test_apply_toindex():
+    # GH 21539
+    tmp = np.array([[1, 1, 1], [1, 0, 0], [1, 1, 0], [1, 1, 0]])
+    tmp_df = pd.DataFrame(tmp)
+    result = tmp_df.apply(lambda x: x[x == 1].index.tolist(), axis=1)
+    expected = pd.Series([[0, 1, 2], [0], [0, 1], [0, 1]])
+    tm.assert_series_equal(result, expected)
+
+    tpm2 = np.array([[5, 1, 3], [1, np.nan, 0], [1, 2, 0], [np.nan, 1, 0]])
+    tmp2_df = pd.DataFrame(tmp2)
```
Comment (gfyoung) [inline pandas/tests/sparse/frame/test_apply.py]: `tmp2` isn't defined (you actually created `tpm2`)

**Comment by uds5501** [comment]
@gfyoung Thanks for pointing that out! That was really stupid by my side

File: pandas/tests/sparse/frame/test_apply.py [hunk 3]
```diff
@@ -90,3 +91,18 @@ def test_applymap(frame):
...
     result = frame.applymap(lambda x: x * 2)
     assert isinstance(result, SparseDataFrame)
+
+
+def test_apply_toindex():
+    # GH 21539
+    tmp = np.array([[1, 1, 1], [1, 0, 0], [1, 1, 0], [1, 1, 0]])
+    tmp_df = pd.DataFrame(tmp)
+    result = tmp_df.apply(lambda x: x[x == 1].index.tolist(), axis=1)
+    expected = pd.Series([[0, 1, 2], [0], [0, 1], [0, 1]])
```
Comment (jreback) [inline pandas/tests/sparse/frame/test_apply.py]: @uds5501 this is still in the wrong file, needs to be in pandas/tests/frame/test_apply.py (its in sparse), and really check whether this actually duplicates current tests.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/sparse/frame/test_apply.py (hunk 3 above)
Comment (uds5501) [reply pandas/tests/sparse/frame/test_apply.py]: @jreback  Oh! okay! thank you for guidance. Will check it rightaway

Comment (uds5501) [reply pandas/tests/sparse/frame/test_apply.py]: @jreback @gfyoung 
 I looked into the frame/test_apply.py and found out that most of the tests are correlating to each other. My test is just unique in application of `index.to_list` and asserting equality with Series, shall I still add these tests in those files?

Comment (jreback) [reply pandas/tests/sparse/frame/test_apply.py]: as I said, I don't this test actually adds anything, this is just return a list in the lambda.

**Pull request closed without merging**